flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/debezium/reader/SnapshotSplitReader.java

```diff
@@ -65,20 +65,24 @@
     private static final Logger LOG = LoggerFactory.getLogger(SnapshotSplitReader.class);
     private final StatefulTaskContext statefulTaskContext;
     private final ExecutorService executor;
+    private final MySqlSnapshotSplit snapshotSplit;
 
     private volatile ChangeEventQueue<DataChangeEvent> queue;
     private volatile boolean currentTaskRunning;
     private volatile Throwable readException;
 
     // task to read snapshot for current split
     private MySqlSnapshotSplitReadTask splitSnapshotReadTask;
-    private MySqlSnapshotSplit currentSnapshotSplit;
     private SchemaNameAdjuster nameAdjuster;
     public AtomicBoolean hasNextElement;
     public AtomicBoolean reachEnd;
 
-    public SnapshotSplitReader(StatefulTaskContext statefulTaskContext, int subtaskId) {
+    public SnapshotSplitReader(
+            StatefulTaskContext statefulTaskContext,
+            MySqlSnapshotSplit snapshotSplit,
+            int subtaskId) {
         this.statefulTaskContext = statefulTaskContext;
+        this.snapshotSplit = snapshotSplit;
         ThreadFactory threadFactory =
                 new ThreadFactoryBuilder().setNameFormat("debezium-reader-" + subtaskId).build();
         this.executor = Executors.newSingleThreadExecutor(threadFactory);
```

Comment:
This will lead to thread leak as there're many snapshot splits, we need avoid this.